index.js

```diff
@@ -16,6 +17,9 @@ module.exports = function gulpPug(options) {
     const data = Object.assign({}, opts.data, file.data || {});
 
     opts.filename = file.path;
+    if (typeof namefunc === 'function') {
+      opts.name = namefunc(file);
```

Comment:
You should move this into the `opts.client` conditional because it is only used for client compilation.  We might as well avoid extra loops when it doesn't effect the output.